src/tuning/kernels/xconvgemm.hpp

```diff
@@ -0,0 +1,186 @@
+
+// =================================================================================================
+// This file is part of the CLBlast project. The project is licensed under Apache Version 2.0. This
+// project loosely follows the Google C++ styleguide and uses a tab-size of two spaces and a max-
+// width of 100 characters per line.
+//
+// Author(s):
+//   Cedric Nugteren <www.cedricnugteren.nl>
+//
+// This file uses the auto-tuner to tune the ConvGemm kernels. These kernels are based on the GEMM
+// direct kernel and will use those parameters, this tuner is just optional to use for advanced
+// users.
+//
+// =================================================================================================
+
+#include <string>
+#include <vector>
+
+#include "utilities/utilities.hpp"
+#include "tuning/tuning.hpp"
+
+namespace clblast {
+// =================================================================================================
+
+// Helper functions
+template <typename T>
+size_t OutputHeight(const Arguments<T> &args) {
+  const auto size = args.height + 2 * args.pad_h;
+  const auto padding = args.dilation_h * (args.kernel_h - 1) + 1;
+  if (size >= padding) { return (size - padding) / args.stride_h + 1; }
+  return 1;
+}
+template <typename T>
+size_t OutputWidth(const Arguments<T> &args) {
+  const auto size = args.width + 2 * args.pad_w;
+  const auto padding = args.dilation_w * (args.kernel_w - 1) + 1;
+  if (size >= padding) { return (size - padding) / args.stride_w + 1; }
+  return 1;
+}
+
+// Settings for this kernel (default command-line arguments)
+TunerDefaults XConvGemmGetTunerDefaults(const int) {
+  auto settings = TunerDefaults();
+  settings.options = {kArgChannels, kArgHeight, kArgWidth, kArgKernelH, kArgKernelW,
+                      kArgNumKernels, kArgBatchCount, kArgFraction};
+  settings.channels = 32;
+  settings.height = 66;
+  settings.width = 66;  // num_patches = 64x64 = 4096
```

Comment:
Is 66 a sensible default here?

Reply:
`global_size` must be multiply of `local_size`, but the tuner directly sets `global_size` to `num_patches` without ceiling it.
Maybe, we also can adjust `global_size` in the tuning framework.

Reply:
Ah so this 66 is computed as 64 + 2 * padding, with the padding being one?

Reply:
It is more complicated.
```
// According to routines/levelx/xconvgemm.cpp
size_h = settings.height + 2 * pad_h
padding_h = dilation_h * (kernel_h - 1) + 1
output_h = (size_h - padding_h) / stride_h + 1

// XConvGemmSetArguments in tuning/kernels/xconvgemm.hpp
padding_h = 0
stride_h = 1
dilation_h = 1
kernel_h = 3

output_h = 64 -> settings.height = 66
```